R/nest.R

```diff
@@ -332,18 +332,9 @@ unnest.data.frame <- function(
 
   cols <- tidyselect::eval_select(enquo(cols), data)
 
-  if (nrow(data) == 0) {
-    for (col in names(cols)) {
-      data[[col]] <- as_empty_df(data[[col]], col = col)
-    }
-  } else {
-    for (col in names(cols)) {
-      data[[col]] <- map(data[[col]], as_df, col = col)
-    }
-  }
```

Comment:
There are some weird edge cases that might break if we remove this. For example, having a mix of vectors and tibbles in the list-col you are trying to unnest currently works like this:

``` r
library(tidyr)

# outer column name of `x` is used to turn `1` into `data.frame(x = 1)`
# then the unchop+unpack is done
df <- tibble(x = list(1, tibble(y = 2)))
unnest(df, x)
#> # A tibble: 2 x 2
#>       x     y
#>   <dbl> <dbl>
#> 1     1    NA
#> 2    NA     2

# again we get `data.frame(x = 1)`, which then row binds with the tibble
# since it also uses `x` as the column name
df <- tibble(x = list(1, tibble(x = 1)))
unnest(df, x)
#> # A tibble: 2 x 1
#>       x
#>   <dbl>
#> 1     1
#> 2     1

# if the column types are different, that would be an error, which leads
# me to believe that this is a "risky" feature to be relying on
df <- tibble(x = list(1, tibble(x = "x")))
unnest(df, x)
#> Error: Can't combine `..1$x` <double> and `..2$x` <character>.
```

The first two would become errors if this code was removed.

Granted, this seems like strange behavior to rely on, so maybe we can just stop supporting it? FWIW `unnest_legacy()` errors in all of these cases, which makes me think that it would be alright to not support this.

Removing this bit of code does greatly improve performance, so it would be nice to do so if possible.

CC @hadley - not sure if you can think of any other edge cases where this code mattered off the top of your head.

Reply:
I have zero recollection 😢 If removing it doesn't breaking any tests and has minimal impact on revdeps, I think it's fine to remove it. `unnest()`ing a mix of vectors and data frames doesn't seem terribly useful, and it does feel like it works by accident currently.